internal/handler/web/users.go

```diff
@@ -90,6 +93,18 @@ func (h *Handler) PostUsersApprovedResearchersImportCsv(ctx *gin.Context) {
 	ctx.Status(http.StatusNoContent)
 }
 
+func EntraUsernameForExternalEmail(email string) string {
```

Comment:
this should return an error not an empty string if it fails by either
- wrong number of parts
- EntraTenantPrimaryDomain is unset